src/app/collective-rewards/user/hooks/useGetGaugesArray.ts

```diff
@@ -1,21 +1,27 @@
-import { useReadContracts } from 'wagmi'
-import { AVERAGE_BLOCKTIME } from '@/lib/constants'
 import { BuilderRegistryAbi } from '@/lib/abis/v2/BuilderRegistryAbi'
-import { AbiFunction, Address } from 'viem'
-import { useGetGaugesLength } from '@/app/collective-rewards/user'
-import { useMemo } from 'react'
+import { AVERAGE_BLOCKTIME } from '@/lib/constants'
 import { BuilderRegistryAddress } from '@/lib/contracts'
+import { useReadBuilderRegistry } from '@/shared/hooks/contracts'
+import { useMemo } from 'react'
+import { AbiFunction, Address } from 'viem'
+import { useReadContracts } from 'wagmi'
 
 const gaugeTypeOptions = ['active', 'halted'] as const
 export type GaugeType = (typeof gaugeTypeOptions)[number]
 
 type FunctionEntry = Extract<(typeof BuilderRegistryAbi)[number], AbiFunction>
-type FunctionName = Extract<FunctionEntry['name'], 'getGaugeAt' | 'getHaltedGaugeAt'>
+type GetAtFunctionName = Extract<FunctionEntry['name'], 'getGaugeAt' | 'getHaltedGaugeAt'>
 
-const gaugeType: Record<GaugeType, FunctionName> = {
+const getAtFunction: Record<GaugeType, GetAtFunctionName> = {
   active: 'getGaugeAt',
   halted: 'getHaltedGaugeAt',
 }
+type GetLengthFunctionName = Extract<FunctionEntry['name'], 'getGaugesLength' | 'getHaltedGaugesLength'>
+
+const getLengthFunction: Record<GaugeType, GetLengthFunctionName> = {
```

Comment:
NIT: I think it may make more sense to have a single record with 2 values for each of the types (active, halted)

Reply:
This is out of scope for this PR.